code/WorkInProgress/parallax.dm

```diff
@@ -0,0 +1,214 @@
+#define PARALLAX_VIEW_WIDTH (WIDE_TILE_WIDTH + 4) // stolen from pali
+#define PARALLAX_VIEW_HEIGHT (SQUARE_TILE_WIDTH + 4)
+
+/datum/parallax_object // stores all the objects for parallax that arent the background
+	var/icon = 'icons/misc/1024x1024.dmi'
+	var/icon_state = "plasma_giant"
+	var/coords = list("X" = 150,"Y" = 150) // coords to stay near
+	var/layer = 1 // layering
+	var/scale = 0.25 // this makes things like planets barely move, higher is faster
+	var/size = 1
+	var/zlevel = 1
+
+	ss14
+		icon = 'icons/obj/backgrounds.dmi'
+		icon_state = "ss14"
+		coords = list("X"=150,"Y" = 130)
+		layer = 2
+		scale = 0.3
+
+	ss12
+		icon = 'icons/obj/backgrounds.dmi'
+		icon_state = "ss12-broken"
+		coords = list("X"=130,"Y" = 150)
+		layer = 2
+		scale = 0.3
+
+	star
+		icon = 'icons/misc/galactic_objects_large.dmi'
+		icon_state = "star-red"
+		coords = list("X"=160,"Y" = 130)
+		scale = 0.125
+		layer = 0
+
+		blue
+			icon = 'icons/misc/galactic_objects_large.dmi'
+			icon_state = "star-blue"
+			coords = list("X"=160,"Y" = 135)
+
+	planet
+		icon = 'icons/misc/512x512.dmi'
+		size = 1
+		scale = 0.5
+		// x1-x2 dont seen to have sprites? what
+		x3
+			icon_state = "moon-green"
+			coords = list("X"=20,"Y" = 260)
+			layer = 2
+			scale = 0.5
+			zlevel = 2
+		x4
+			icon = 'icons/obj/large/160x160.dmi'
+			icon_state = "bigasteroid_1"
+
+		x5
+			icon_state = "moon-chunky"
+
+		x15
+			icon_state = "moon-ice"
+
+		domus_dei
+			icon_state = "domusDei"
+
+		quadriga
+			icon_state = "quadriga"
+
+		mundus
+			icon_state = "mundus"
+
+		iustitia
+			icon_state = "iustitia"
+
+		iudicium
+			icon_state = "iudicium"
+
+		fortuna
+			icon_state = "fortuna"
+
+		fatuus
+			icon_state = "fatuus"
+
+		magus
+			icon_state = "magus"
+
+		regis
+			icon_state ="regis"
+
+		amantes
+			icon_state = "amantes"
+
+		antistes
+			icon_state = "antistes"
+
+// wow there are so many different objects in space
+
+/datum/hud/parallax
+	// this is a hud that when given to a client will add parallax to their view.
+	var/list/parallax_objects = list()
+	var/list/parallax_settings = list()
+	var/list/parallax_coords = list(list())
+	var/list/parallax_zlevel = list()
+	var/list/scale = list()
+	var/list/size = list()
+
+	var/mob/master
+	var/active = TRUE
+
+	var/atom/movable/screen/hud/background
+
+	New(M)
+		..()
+		if(isnull(M))
+			CRASH("parallax HUD created with no master")
+		master = M
+		clients += master.client
+		SPAWN(0)
+			background = create_screen("background", "Space", 'icons/effects/overlays/parallaxBackground.dmi', "background", "1,1", HUD_LAYER-1)
+			background.transform = matrix(0,0,0,0,0,0)
+			background.screen_loc = master?.client?.view ? "4,1" : "1,1"
```

Comment:
whats this

Reply:
that is the setup for the background used for this, it didnt need some of the stuff that the planets needed so i put it there and left it out of the planet list

Reply:
why is it in a spawn 0

Reply:
i dont even know why i put that there, removed it